circuits/integration/mpc_circuits/match.rs

```diff
@@ -60,56 +59,65 @@ fn test_match_no_match(test_args: &IntegrationTestArgs) -> Result<(), String> {
         .map_err(|err| format!("Error allocating balance2 in the network: {:?}", err))?;
 
     // Build the test cases for different invalid match pairs
-    let mut test_cases: Vec<Vec<u64>> = vec![
+    let test_cases: Vec<(Order, u64)> = vec![
         // Quote mints different
-        vec![
-            sel!(0, 1),   /* quote_mint */
-            2,            /* base_mint */
-            sel!(0, 1),   /* side */
-            sel!(20, 30), /* amount */
-            10,           /* price */
-        ],
+        (
+            Order {
+                quote_mint: sel!(0u8, 1u8).into(),
+                base_mint: 2u8.into(),
+                side: sel!(OrderSide::Buy, OrderSide::Sell),
+                worst_case_price: FixedPoint::from_integer(sel!(15, 5)),
+                amount: sel!(20, 30),
+                timestamp: 0, // unused
+            },
+            10, /* execution_price */
+        ),
         // Base mints different
-        vec![
-            1,            /* quote_mint */
-            sel!(1, 2),   /* base_mint */
-            sel!(0, 1),   /* side */
-            sel!(20, 30), /* amount */
-            10,           /* price */
-        ],
-        // Both orders on the same side (buy)
-        vec![
-            1,            /* quote_mint */
-            2,            /* base_mint */
-            0,            /* side (both buy) */
-            sel!(20, 30), /* amount */
-            10,           /* price */
-        ],
+        (
+            Order {
+                quote_mint: 1u8.into(),
+                base_mint: sel!(0u8, 1u8).into(),
+                side: sel!(OrderSide::Buy, OrderSide::Sell),
+                worst_case_price: FixedPoint::from_integer(sel!(15, 5)),
+                amount: sel!(20, 30),
+                timestamp: 0, // unused
+            },
+            10, /* execution_price */
+        ),
+        // Orders on the same side (buy side)
+        (
+            Order {
+                quote_mint: 1u8.into(),
+                base_mint: 2u8.into(),
+                side: OrderSide::Buy,
+                worst_case_price: FixedPoint::from_integer(15),
```

Comment:
Should the `worst_case_price` be set differently for each relayer here, i.e. `FixedPoint::from_integer(sel!(5, 15))`?

Just wanna make sure this test fails because of the orders being on the same side and not because of a price protection violation

Reply:
This test doesn't interact with this field, plus it makes sense for the worst case prices to be directionally the same since the orders are on the same side of the market.